[WebGPU] Unhandled texture formats in wgslFormat

tadeuzagallo opened the pull request:
#### beec05729a177860ead4a841f4902749a37ffd5f
<pre>
[WebGPU] Unhandled texture formats in wgslFormat
<a href="https://bugs.webkit.org/show_bug.cgi?id=298315">https://bugs.webkit.org/show_bug.cgi?id=298315</a>
<a href="https://rdar.apple.com/159751366">rdar://159751366</a>

Reviewed by Mike Wyrzykowski.

`texture-formats-tier1` was added in 299147@main, but the new formats weren&apos;t added
to `wgslFormat`, which resulted in a crash in debug builds or an incorrect value being
returned in release.

* LayoutTests/fast/webgpu/nocrash/fuzz-x-expected.txt: Added.
* LayoutTests/fast/webgpu/nocrash/fuzz-x.html: Added.
* Source/WebGPU/WebGPU/ShaderModule.mm:
(WebGPU::wgslFormat):

Canonical link: <a href="https://commits.webkit.org/299548@main">https://commits.webkit.org/299548@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/ca17d18d7daa02070458e95b79da5ecbb87f1a76

| Misc | iOS, visionOS, tvOS & watchOS  | macOS  | Linux |  Windows | Apple Internal |
| ----- | ---------------------- | ------- |  ----- |  --------- | ------ |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/119378 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/131/builds/39065 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/138/builds/29720 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/5/builds/125626 "Built successfully") | [✅ 🛠 win](https://ews-build.webkit.org/#/builders/59/builds/71449 "Built successfully") | [✅ 🛠 ios-apple](https://ews-bridge.webkit.apple.com/builds/sw/T-276/92699e7b-e2c4-41dd-8b79-0fc7c00d5b52/95a5a5de-4848-4d3d-8c56-2ed4090719c4) 
| | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/130/builds/39762 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/123/builds/47646 "Built successfully") | [✅ 🧪 wpe-wk2](https://ews-build.webkit.org/#/builders/34/builds/90707 "Passed tests") | [  ~~🧪 win-tests~~](https://ews-build.webkit.org/#/builders/60/builds/60025 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [✅ 🛠 mac-apple](https://ews-bridge.webkit.apple.com/builds/sw/T-276/92699e7b-e2c4-41dd-8b79-0fc7c00d5b52/ce6dc632-9be7-49cd-ba88-709d2ada9350) 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/122330 "Passed tests") | [✅ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/132/builds/31730 "Passed tests") | [✅ 🧪 api-mac](https://ews-build.webkit.org/#/builders/18/builds/107030 "Passed tests") | [✅ 🧪 api-wpe](https://ews-build.webkit.org/#/builders/41/builds/71163 "Passed tests") | | [✅ 🛠 vision-apple](https://ews-bridge.webkit.apple.com/builds/sw/T-276/92699e7b-e2c4-41dd-8b79-0fc7c00d5b52/7cd02153-09a3-4b04-a140-f7ba5bd25411) 
| | [✅ 🧪 ios-wk2-wpt](https://ews-build.webkit.org/#/builders/133/builds/30765 "Passed tests") | [✅ 🧪 mac-wk1](https://ews-build.webkit.org/#/builders/135/builds/25139 "Passed tests") | [✅ 🛠 wpe-cairo](https://ews-build.webkit.org/#/builders/65/builds/69279 "Built successfully") | | 
| | [✅ 🧪 api-ios](https://ews-build.webkit.org/#/builders/13/builds/101184 "Passed tests") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/136/builds/25330 "Passed tests") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/2/builds/128623 "Built successfully") | | 
| | [✅ 🛠 vision](https://ews-build.webkit.org/#/builders/128/builds/46296 "Built successfully") | [❌ 🧪 mac-AS-debug-wk2](https://ews-build.webkit.org/#/builders/122/builds/35034 "Found 1 new test failure: http/tests/websocket/web-socket-loads-captured-in-per-page-domains.html (failure)") | [✅ 🧪 gtk-wk2](https://ews-build.webkit.org/#/builders/1/builds/99282 "Passed tests") | | 
| | [✅ 🛠 vision-sim](https://ews-build.webkit.org/#/builders/121/builds/46661 "Built successfully") | [✅ 🧪 mac-wk2-stress](https://ews-build.webkit.org/#/builders/8/builds/103229 "Passed tests") | [✅ 🧪 api-gtk](https://ews-build.webkit.org/#/builders/21/builds/99074 "Passed tests") | | 
| [✅ 🛠 🧪 merge](https://ews-build.webkit.org/#/builders/19/builds/25180 "Built successfully and passed tests") | [✅ 🧪 vision-wk2](https://ews-build.webkit.org/#/builders/126/builds/44520 "Passed tests") | [✅ 🧪 mac-intel-wk2](https://ews-build.webkit.org/#/builders/137/builds/22524 "Passed tests") | [  ~~🛠 playstation~~](https://ews-build.webkit.org/#/builders/134/builds/42863 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | | 
| | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/127/builds/46159 "Built successfully") | [✅ 🛠 mac-safer-cpp](https://ews-build.webkit.org/#/builders/120/builds/51859 "Built successfully") | | | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/125/builds/45624 "Built successfully") | | | | 
| | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/129/builds/48974 "Built successfully") | | | | 
| | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/124/builds/47311 "Built successfully") | | | | 
<!--EWS-Status-Bubble-End-->

webkit-early-warning-system commented:
Starting EWS tests for https://github.com/WebKit/WebKit/commit/4a18c008049ab7302ac446429563e951e1802418. Live statuses available at the PR page, https://github.com/WebKit/WebKit/pull/50236

mwyrzykowski commented on LayoutTests/fast/webgpu/nocrash/fuzz-298315-expected.txt:
```diff
@@ -0,0 +1,4 @@
+CONSOLE MESSAGE: 0,0
+CONSOLE MESSAGE: 414141,0,0,0,66000000,0,0,0
```
for some reason EWS is showing:
```
CONSOLE MESSAGE: 414141,0,0,0,66000002,0,0,0
```
or
```
CONSOLE MESSAGE: 414141,0,0,0,66000001,0,0,0
```
instead

mwyrzykowski replied:
hmm this would mean that the provoking vertex is not always the first vertex_id

mwyrzykowski replied:
oh that's due to `primitive: {topology: 'point-list'},` - test is using points and not triangles

That makes sense then, in which case the test expectations differ due to a race condition

tadeuzagallo committed:
[WebGPU] Unhandled texture formats in wgslFormat
https://bugs.webkit.org/show_bug.cgi?id=298315
rdar://159751366

Reviewed by Mike Wyrzykowski.

`texture-formats-tier1` was added in 299147@main, but the new formats weren't added
to `wgslFormat`, which resulted in a crash in debug builds or an incorrect value being
returned in release.

* LayoutTests/fast/webgpu/nocrash/fuzz-x-expected.txt: Added.
* LayoutTests/fast/webgpu/nocrash/fuzz-x.html: Added.
* Source/WebGPU/WebGPU/ShaderModule.mm:
(WebGPU::wgslFormat):

Canonical link: https://commits.webkit.org/299548@main
